app/models/task_evaluator.rb

```diff
@@ -3,4 +3,15 @@ class TaskEvaluator < ApplicationRecord
   belongs_to :evaluator
 
   validates :evaluator, uniqueness: { scope: :task_id }
+  validate :matching_task_evaluator
```

Comment:
Maybe a tiny renaming to make it more straightforward:

```suggestion
  validate :evaluator_compatibility
```